### PR TITLE
[Adminutils] add emoji from message

### PR DESCRIPTION
**Pull request by kablekompany**
Steal emoji from a message using message.id
(probably not the best logic in mirroring your own but sharing nonetheless!

### CODE REVIEW

File: adminutils/adminutils.py [hunk 1]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
         else:
             await ctx.tick()
 
+    @emoji.command(name="steal")
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
    @emoji.command(name="message", aliases=["steal"])
```

File: adminutils/adminutils.py [hunk 2]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+    async def emote_steal(self, ctx, name: str, message_id: int, *roles: discord.Role):
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
    async def emote_steal(self, ctx, name: str, message_id: discord.Message, *roles: discord.Role):
```

File: adminutils/adminutils.py [hunk 3]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+        """
+        Steal an emoji from a message ID and add it to your server.
+        Role lock the emoji if you wish (use double quotes if role name has spaces)
+
+        Examples:
+            `[p]emoji steal thisReallyCoolName 743291269471797248`
+            `[p]emoji steal thisReallyCoolName 743291269471797248 RoleName`
+        """
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
        """
        Add an emoji from a specified message and add it to your server.
        Use double quotes if role name has spaces

        Examples:
            `[p]emoji message Example 162379234070467641`
            `[p]emoji message RoleBased 162379234070467641 EmojiRole`
        """
```

File: adminutils/adminutils.py [hunk 4]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+        channel = ctx.channel
+        message = await channel.fetch_message(message_id)
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
```

Not needed with message converter

File: adminutils/adminutils.py [hunk 5]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+        emoji = "".join(x for x in dig_up_emoji if x.isdigit())
+        check_animated = dig_up_emoji_check.endswith("<a")
+        if check_animated is True:
+            format_image = "gif"
+        else:
+            format_image = "png"
+        url = f"https://cdn.discordapp.com/emojis/{emoji}.{format_image}"
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url) as r:
+                data = await r.read()
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
        async with self.session.get(url) as r:
            data = await r.read()
```

Cog already have a session for that.

This also will throw an `check your logs` error if emoji is unable to unload, this better to be handled too. Look at how `emoji_add` handles that

File: adminutils/adminutils.py [hunk 6]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+        message = await channel.fetch_message(message_id)
+        dig_up_emoji_check = message.content.split(":")[-3]
+        check_animated = dig_up_emoji_check.endswith("<a")
+        dig_up_emoji = message.content.split(":")[-1]
+        emoji = "".join(x for x in dig_up_emoji if x.isdigit())
+        check_animated = dig_up_emoji_check.endswith("<a")
+        if check_animated is True:
+            format_image = "gif"
+        else:
+            format_image = "png"
```
Comment (fixator10) [inline adminutils/adminutils.py]: This better to be redone with regex.

In example, as it done in [TrustyCogs' notsobot converter](https://github.com/TrustyJAID/Trusty-cogs/blob/a3e931bc6227645007b37c3f4f524c9fc9859686/notsobot/converter.py#L30-L36).

File: adminutils/adminutils.py [hunk 7]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+            async with session.get(url) as r:
+                data = await r.read()
+        try:
+            await ctx.guild.create_custom_emoji(
+                name=name,
+                image=data,
+                roles=roles,
+                reason=get_audit_reason(
+                    ctx.author,
+                    ("Restricted to roles: {}").format(
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
                    _("Restricted to roles: {}").format(
```
i18n support

File: adminutils/adminutils.py [hunk 8]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+
+            await ctx.send(embed=e)
+        except discord.InvalidArgument:
+            await ctx.send(
+                "Make sure there's an emoji in that messageID and you're running this command correctly (`[p]emoji steal nameTheEmote <msgID>`) Extra numbers in the message will throw this error too."
+            )
+            return
+        except discord.HTTPException:
+            await ctx.send("You are at your emoji limit it would seem.")
+            return
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
        except discord.HTTPException as e:
            await ctx.send(chat.error(_("An error occurred on adding an emoji: {}").format(e)))
```
HTTPException can happen not only on hitting emoji limit.
About emoji limit - perhaps even better handle this at start of command via check for count of current server emojis and total available emojis (`guild.emoji_limit`)

File: adminutils/adminutils.py [hunk 9]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+            msg = "Emote added, fam!"
+            e = discord.Embed(title=msg, colour=discord.Colour.dark_green())
+            e.set_author(name=ctx.author.name, icon_url=ctx.author.avatar_url)
+            e.set_image(url=url)
+            e.set_thumbnail(url=ctx.guild.icon_url)
+
+            await ctx.send(embed=e)
+        except discord.InvalidArgument:
+            await ctx.send(
+                "Make sure there's an emoji in that messageID and you're running this command correctly (`[p]emoji steal nameTheEmote <msgID>`) Extra numbers in the message will throw this error too."
```
Comment (fixator10) [inline adminutils/adminutils.py]:
```suggestion
                _("This image type is not supported anymore or Discord returned incorrect data. Try again later.")
```

File: adminutils/adminutils.py [hunk 10]
```diff
@@ -183,6 +183,62 @@ async def emoji_add(self, ctx, name: str, url: str, *roles: discord.Role):
...
+                    if roles
+                    else None,
+                ),
+            )
+            await ctx.tick()
+            msg = "Emote added, fam!"
+            e = discord.Embed(title=msg, colour=discord.Colour.dark_green())
+            e.set_author(name=ctx.author.name, icon_url=ctx.author.avatar_url)
+            e.set_image(url=url)
+            e.set_thumbnail(url=ctx.guild.icon_url)
```
Comment (fixator10) [inline adminutils/adminutils.py]: Other commands doesn't have any message on success.
```suggestion
```

**Review by fixator10 — CHANGES_REQUESTED** [review]

**Comment by kablekompany** [comment]
Cant' appease black for some reason but requested changes were committed/performed

**Review by fixator10 — APPROVED** [review]

**Comment by fixator10** [comment]
Thanks for contribution

**Comment by kablekompany** [comment]
thanks for your guidance and patience. Cheers!

**Pull request merged**